mesonbuild/msetup.py

```diff
@@ -299,6 +300,10 @@ def _finalize_devenv(self, b: build.Build, intr: interpreter.Interpreter) -> Non
             devenv = mod.get_devenv()
             if devenv:
                 b.devenv.append(devenv)
+        # Dump devenv into a file that can be used by vscode's launch.json
+        dump_path = os.path.join(self.build_dir, 'vscode.env')
```

Comment:
This creates a brand new reserved filename in the build directory root?

Also since apparently this is vscode-specific maybe only create it if vscode is installed.

Reply:
> This creates a brand new reserved filename in the build directory root?

Right, maybe it should be prefixed with `meson-`.

> Also since apparently this is vscode-specific maybe only create it if vscode is installed.

I'm not sure what other IDE could benefit from this feature, maybe that's more generic than just vscode?

Reply:
Maybe only generate vscode.env if `.vscode/launch.json` file exists?

Reply:
Sure, maybe. Alternatively it could be created by the vscode extension.

But regardless -- I say this is vscode specific because the filename has the word "vscode" in it, which strikes me as exceedingly specific. ;) If the idea was to be generic then it would have to be, well, generic. :)

Reply:
> Sure, maybe. Alternatively it could be created by the vscode extension.

That's actually a good point, as long as we have the CLI to dump to file, it can be done by vscode extension indeed. I'm just worried that if user reconfigure the project outside of vscode, the env file might not be up to date. But I guess that's true for every introspection data the extension uses, it could monitor coredata.dat and regenerate automatically.

> But regardless -- I say this is vscode specific because the filename has the word "vscode" in it, which strikes me as exceedingly specific. ;) If the idea was to be generic then it would have to be, well, generic. :)

Right, I named it like that because I only know about vscode that needs it, but if someone knows other IDEs that could use it, then we could have a more generic name. But as said above, I think I'll drop the commit that generate the file automatically and leave that to vscode extension.